ValueUtils/ReflectionHelper.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ValueUtils
+{
+    static class ReflectionHelper
+    {
+        public static IEnumerable<Type> WalkMeaningfulInheritanceChain(Type type)

```

Comment:
This method crashes with a `NullReferenceException` when `type` is an interface type. I assume this is an unsupported scenario anyway, but perhaps there should be a more specific error.